sdk/eventhub/eventhubs-checkpoint-blob/src/blobPartitionManager.ts

```diff
@@ -0,0 +1,209 @@
+   *
+   * @param eventHubName The event hub name.
+   * @param consumerGroupName The consumer group name.
+   * @return Partition ownership details of all the partitions that have/had an owner..
+   */
+  async listOwnership(
+    eventHubName: string,
+    consumerGroupName: string
+  ): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
+    try {
+      for await (const blob of this._containerClient.listBlobsFlat({
+        include: ["metadata"],
+        prefix: `${eventHubName}/${consumerGroupName}/`
+      })) {
+        const blobPath = blob.name.split("/");
+        const blobName = blobPath[blobPath.length - 1];
+        const partitionOwnership: PartitionOwnership = {
+          eventHubName,
+          consumerGroupName,
+          ownerId: blob.metadata!.ownerid,
+          partitionId: blobName,
+          offset: blob.metadata && blob.metadata.offset ? parseInt(blob.metadata.offset) : -1,
+          sequenceNumber:
+            blob.metadata && blob.metadata.sequencenumber
+              ? parseInt(blob.metadata.sequencenumber)
+              : -1,
+          lastModifiedTimeInMS:
+            blob.properties.lastModified && Date.parse(blob.properties.lastModified.toISOString()),
+          eTag: blob.properties.etag,
+          ownerLevel: 0 // this needs to be removed from eventhubs
+        };
+        partitionOwnershipArray.push(partitionOwnership);
+      }
+      return partitionOwnershipArray;
+    } catch (err) {
+      log.error(`Error ocuured while fetching the list of blobs.`, err);
+      throw new Error(`Error ocuured while fetching the list of blobs. \n${err}`);
+    }
+  }
+
+  /**
+   * Claim ownership of a list of partitions. This will return the list of partitions that were owned
+   * successfully.
+   *
+   * @param partitionOwnership The list of partition ownership this instance is claiming to own.
+   * @return A list partitions this instance successfully claimed ownership.
+   */
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    let partitionOwnershipArray: PartitionOwnership[] = [];
+    for (const ownership of partitionOwnership) {
+      const blobName = `${ownership.eventHubName}/${ownership.consumerGroupName}/${ownership.partitionId}`;
+      try {
+        let updatedBlobResponse;
+        const blobClient = this._containerClient.getBlobClient(blobName);
+        if (ownership.eTag) {
+          updatedBlobResponse = await blobClient.setMetadata(
+            {
+              OwnerId: ownership.ownerId ? ownership.ownerId : "",
+              SequenceNumber: ownership.sequenceNumber ? ownership.sequenceNumber.toString() : "",
+              Offset: ownership.offset ? ownership.offset.toString() : ""
+            },
+            {
+              blobAccessConditions: {
+                modifiedAccessConditions: {
+                  ifMatch: ownership.eTag
```

Comment:
These two if else blocks are different. In if condition I'm checking if `eTag` exist then just call `setMetadata()` method otherwise call `upload()` method (first time if we call this method, it will call `upload()`).

I think `upload()` would be more expensive as compared to `setMetadata()`. That's why I didn't combine these two.

Reply:
I missed that you weren't using `upload` for both cases, you're right.